### ISSUE TITLE
Remove Prometheus from the list of default middleware

### ISSUE THREAD

**Issue by CaselIT** [Bogdanp/dramatiq#345]
# Issues

Currently (1.9.0) the prometheus middleware is among the default middleware that are used if an user does not specify any.
Also by default it connects to the `0.0.0.0` address, making the server accessible on the external network.
https://github.com/Bogdanp/dramatiq/blob/13e8d626c4d0250a92a436324b0039204a3cf706/dramatiq/middleware/prometheus.py#L37

I think it may make sense to remove it from the default middleware, since it offers functionality that is not likely to be used by the majority of the users.

At least it should only connect to `localhost` by default and not to `0.0.0.0`.

## What OS are you using?

windows (not relevant here

## What version of Dramatiq are you using?

1.9.0

## What did you expect would happen?

prometheus is an optional middleware

## What happened?

prometheus is strarted by default and exposes information on 0.0.0.0

**Comment by Bogdanp** [comment]
I plan to do this as part of a 2.0 release with small breaking changes like this one at some point in the future.

**Comment by CaselIT** [comment]
Ok, thanks!

**Comment by jwilges** [comment]
Hey @Bogdanp, as a transitional step to resolve this in the near term, would you be open to a minimal patch not gated by a lofty move to v2.0.0 if it keeps the same default behavior (i.e. `default_middleware = [Prometheus, ...]`) when `prometheus_client` is available--thereby not breaking 1.x backward compatibility--while gracefully omitting the middleware when the client module isn't available?

So far, this library is great overall, but it's a mildly sore spot to have the dependency forced on all installs without a path around it aside from forking. That's not to say that the Prometheus middleware isn't nice to have in the core package, just that it's unfortunate it isn't handled as gracefully as e.g. `RabbitmqBroker` falling back to `RedisBroker` in `get_broker`.

I can totally appreciate that you wouldn't want to test for its presence by importing it, but I do think a gentle `find_spec()` type of approach like I've shown in my forked branch here could be acceptable:
[dramatiq/middleware/__init__.py](https://github.com/Bogdanp/dramatiq/compare/master...jwilges:dramatiq:jwilges/minimal_optional_prometheus#diff-54634011579a78d90a3b90c6ac72b3f6e11a14078160331717f8fa7d58f08634R45-R54)

Long term, v2.0.0 of course could still make a choice to evolve to deprecate it as a default middleware if desired, but this type of interim approach I think could be a nice stepping stone.

I'd be happy to open a pull request and update documentation/tests as well, if you're okay with the idea.

**Comment by Bogdanp** [comment]
@jwilges that sounds like a good idea to me. Feel free to open a PR in that direction!
